controllers/resources/status.go

```diff
@@ -62,6 +65,9 @@ func UpdateStatusNodeHealthy(node *corev1.Node, nhc *remediationv1alpha1.NodeHea
 	delete(nhc.Status.InFlightRemediations, node.GetName())
 	for i, _ := range nhc.Status.UnhealthyNodes {
 		if nhc.Status.UnhealthyNodes[i].Name == node.GetName() {
+			for _, remediation := range nhc.Status.UnhealthyNodes[i].Remediations {
```

Comment:
I don't expect to have many `.Remediations` in the list, but this still increases the complexity.
Since we are working on a single node, we could store the list of its remediations and deal with them outside the loop, I think.

something like
```
func UpdateStatusNodeHealthy(node *corev1.Node, nhc *remediationv1alpha1.NodeHealthCheck, log *logr.Logger) {
	delete(nhc.Status.InFlightRemediations, node.GetName())
	var remediations []*remediationv1alpha1.Remediation
	for i := range nhc.Status.UnhealthyNodes {
		if nhc.Status.UnhealthyNodes[i].Name == node.GetName() {
			remediations = nhc.Status.UnhealthyNodes[i].Remediations
			nhc.Status.UnhealthyNodes = append(nhc.Status.UnhealthyNodes[:i], nhc.Status.UnhealthyNodes[i+1:]...)
			break
		}
	}
	for _, r := range remediations {
		metrics.ObserveNodeHealthCheckUnhealthyNodeOngoingRemediation(node.GetName(), r.Resource.Namespace, r.Resource.Kind, 0)
	}
}
```